heron/scheduler-core/src/java/com/twitter/heron/scheduler/UpdateTopologyManager.java

```diff
@@ -152,6 +152,9 @@ private void updateTopology(final PackingPlans.PackingPlan existingProtoPackingP
     // deactivate and sleep
     if (initiallyRunning) {
       deactivateTopology(stateManager, updatedTopology);
+      // Update the topology since the state should have changed from RUNNING to PAUSED
+      updatedTopology =
```

Comment:
Great catch! Could we instead change `deactivateTopology` to return the new topology, and push the fetch into that method?